api/envoy/extensions/filters/http/gzip/v3/gzip.proto

```diff
@@ -75,5 +75,37 @@ message Gzip {
   // Set of configuration parameters common for all compression filters. If this field is set then
   // the fields `content_length`, `content_type`, `disable_on_etag_header` and
   // `remove_accept_encoding_header` are ignored.
-  compressor.v3.Compressor compressor = 10;
+  Compressor compressor = 10;
```

Comment:
Technically since we already froze v3, someone might be using this already. Is the local clone wire compatible with what came before? This is a little shady but I think I'm probably OK with this given how new v3 is. cc @envoyproxy/api-shepherds

Reply:
Yes, the local clone is just a copy/paste from v2. I tried to use v2 first, and it even compiled and worked, but docs generating got broken.

Reply:
I'd prefer on principle not to break v3.. is it really so bad to keep the file where it was before?

Reply:
If to leave it untouched then having these lines in config:
```
          ...
          http_filters:
          - name: envoy.filters.http.gzip
            config:
              compressor:
                content_length: 50
          - name: envoy.filters.http.router
            config: {}
```

causes Envoy fail to start with
```
Proto constraint validation failed (GzipValidationError.Compressor: ["embedded message failed validation"] | caused by CompressorValidationError.CompressorLibrary: ["value is required"])
```
Though in the context of the Gzip http-filter `CompressorLibrary` is hardcoded already: it's gzip and can't be anything else.

Reply:
Is it plausible this existing API is in use by somebody at this point? If so, we need to follow the API stability rules and not break for v3.

You might need to live with deprecating the existing scheme, which might be over-verbose and redundant, and introducing the new one side-by-side, if that is the situation.

Reply:
@htuch The problem is not about being redundant: existing configurations will be broken, because this PR introduces a new mandatory field to `compressor.v3.Compressor` - `compressor_library`.
I guess the only option left is to update the generic filter to use v4alpha messages. Or maybe there's a better way?

Reply:
Maybe don't make it required and then have Envoy implementation code that can raise errors if missing for generic compressor?

Reply:
Oh, right. I was too determent to make it required at the proto definition level.
Thanks! Now the frozen API isn't updated.